### PR TITLE
Add basic cypress tests for the editor and citation-plugin

### PR DESCRIPTION
**Pull request by elpoelma**
## Overview
This PR adds three additional cypress tests:
- A test which checks if basic typing in the editor works
- A test which checks if the citation-plugin modal displays at least 1 result
- A test which checks if the filtering of the citation-plugin works as expected

### How to test/reproduce
- Start the dummy app
- Start the cypress suite
- Ensure that all new test runs are successful

### CODE REVIEW

File: cypress/e2e/citation-plugin.cy.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    cy.visit('test.html');
+    cy.get('.say-editor-hints__insert').contains('button', 'Insert').click();
+    cy.get('.say-editor-hints__insert')
+      .contains('button', 'Insert reference')
+      .click();
+    cy.get('.citaten--main-container')
+      .find('.au-c-main-container__content')
+      .find('ul')
+      .as('list');
```
Comment (piemonkey) [inline cypress/e2e/citation-plugin.cy.js:13]: I thought the idea of `as()` was to mark elements for use between functions, for example when running set-up code. I think in this case we can just store the result of `find()` in a const and then check that.

Comment (elpoelma) [reply cypress/e2e/citation-plugin.cy.js:13]: Yeah, I thought so too, but apparently it is an anti-pattern: https://docs.cypress.io/guides/references/best-practices#Assigning-Return-Values. As cypress works in an asynchronous way.

File: cypress/e2e/citation-plugin.cy.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    cy.get('@list')
+      .children('li')
+      .each((el) => {
+        cy.wrap(el).contains('bos', { matchCase: false });
+      });
+  });
+});
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply cypress/e2e/citation-plugin.cy.js:36]: This is indeed a very welcome improvement :). I agree that we should have as little code duplication as possible (as long at is readable). Also nice that you found a way to set the browser locale :)

File: cypress/e2e/citation-plugin.cy.js (hunk 1 above)
Comment (piemonkey) [reply cypress/e2e/citation-plugin.cy.js]: Ah, good to know. Not sure where I got that impression from.

**Comment by elpoelma** [comment]
- Added the locale and viewport configuration to the e2e support file, so it runs before all tests
- Reduce additional duplication in both main and citation test modules.

**Review by piemonkey — APPROVED** [review]

**Pull request merged**